recipes/libsafec/all/conanfile.py

```diff
@@ -0,0 +1,93 @@
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    build_requires = "autoconf/2.69", "libtool/2.4.6"
+    exports_sources = "patches/*"
+
+    __autotools = None
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    @property
+    def _supported_compiler(self):
+        compiler = self.settings.compiler
+        version = tools.Version(self.settings.compiler.version)
+        if compiler == "Visual Studio":
+            return False
+        if compiler == "gcc" and version < "5":
+            return False
+        return True
+
+    def configure(self):
+        if not self._supported_compiler:
+            raise ConanInvalidConfiguration(
+                "libsafec doesn't support compiler: {} on OS: {}.".format(
+                    self.settings.compiler, self.settings.os))
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob("safeclib-*")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    @property
+    def _autotools(self):
+        if self.__autotools:
+            return self.__autotools
+        self.run("autoreconf -fiv", run_environment=True)
+        self.__autotools = AutoToolsBuildEnvironment(self)
+        if self.options.shared:
+            args = ["--enable-shared", "--disable-static"]
+        else:
+            args = ["--disable-shared", "--enable-static"]
+        args.extend(["--disable-doc", "--disable-dependency-tracking"])
+        if self.settings.build_type in ("Debug", "RelWithDebInfo"):
+            args.append("--enable-debug")
+        self.__autotools.configure(args=args)
+        return self.__autotools
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "Makefile.am"), "_@SAFEC_API_VERSION@", "")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "src", "Makefile.am"), "-@SAFEC_API_VERSION@", "")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "tests", "Makefile.am"), "-@SAFEC_API_VERSION@", "")
+        with tools.chdir(self._source_subfolder):
+            self._autotools.make()
+
+    def package(self):
+        with tools.chdir(self._source_subfolder):
+            self._autotools.install()
+        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
+        with tools.chdir(os.path.join(self.package_folder, "lib")):
+            tools.rmdir("pkgconfig")
+            tools.remove_files_by_mask(".", "*.la")
+
+    def package_info(self):
```

Comment:
You are missing hte pck_config names https://github.com/rurban/safeclib/blob/7ef68891f130e257bf7e442b0dce626df7dcf068/libsafec.pc.in#L11

Reply:
The pkg-config name is `libsafec`, matches Conan package name, so no extra setting is required.

Reply:
Oh sorry, no explanation from me...

So the pc generator will not be created unless explicitly set in the package info

So in CCI we are just trying to stay ahead of that

Reply:
I still don't understand. Here is how pc-file generates:

~~~
$ conan install libsafec/3.6.0@ -g pkg_config && cat libsafec.pc
...
Generator pkg_config created libsafec.pc
prefix=/tmp_proj/adklimki/conan-center-test/.conan/data/libsafec/3.6.0/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709
libdir=${prefix}/lib
includedir=${prefix}/include/libsafec

Name: libsafec
Description: This library implements the secure C11 Annex K[^5] functions on top of most libc implementations, which are missing from them.
Version: 3.6.0
Libs: -L"${libdir}" -lsafec-3.6.0  -Wl,-rpath,"${libdir}"
Cflags: -I"${includedir}"
~~~

Reply:
https://github.com/conan-io/conan/issues/7560

Reply:
We'd prefer to be explicit

Reply:
> [conan-io/conan#7560](https://github.com/conan-io/conan/issues/7560)

That's for components and we don't use components here.

Reply:
It's still nice to be explicit, if not for the convenience of maintainers.
When it is present, a maintainer knows that it is put there explicitly and thus correct.

Reply:
To clarify, the suggestion is to add this line?
~~~python
    self.cpp_info.names["pkg_config"] = "libsafec"
~~~

Reply:
I guess the value for this would be if we decide to change the package name this stays the same... Still feels like unnecessary code duplication.
https://github.com/conan-io/conan-center-index/pull/4434/commits/178a6bf86be10d53efe82733501060c5af329cf6

Reply:
It's a way of documenting that the pkg config name is set..
Thanks for making the change.